server/pulp/server/async/tasks.py

```diff
@@ -271,7 +271,8 @@ def apply_async(self, *args, **kwargs):
         ### This is a temporary workaround for https://github.com/celery/celery/issues/2216 ###
         queue_name_to_save = queue
         if kwargs.get('exchange', None) == DEDICATED_QUEUE_EXCHANGE:
-            kwargs['queue'] = queue.rstrip('.dq')
+            # Strip off '.dq' from the end of the queue name
+            kwargs['queue'] = queue[:-3]
```

Comment:
I am very, very confident that when the exchange specifies the dedicated queue exchange, the queue name contains a .dq at the end of it introduced in the assigned_queue name on [L#126](https://github.com/bmbouter/pulp/blob/1179463/server/pulp/server/async/tasks.py#L126). Pulp only uses the dedicated queue exchange for tasks dispatched using appy_async_with_reservation.